pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -165,10 +170,21 @@ func newRepositoryWithClient(
 	repo distribution.Repository,
 	options map[string]interface{},
 ) (distribution.Repository, error) {
+	// TODO: Deprecate this environment variable.
 	registryAddr := os.Getenv(DockerRegistryURLEnvVar)
 	if len(registryAddr) == 0 {
-		return nil, fmt.Errorf("%s is required", DockerRegistryURLEnvVar)
+		registryAddr = os.Getenv(OpenShiftDefaultRegistry)
+	} else {
+		context.GetLogger(ctx).Infof("DEPRECATED: %q is deprecated, use the %q instead", DockerRegistryURLEnvVar, OpenShiftDefaultRegistry)
+	}
+	if len(registryAddr) == 0 {
+		if len(os.Getenv("DOCKER_REGISTRY_SERVICE_HOST")) > 0 && len(os.GetEnv("DOCKER_REGISTRY_SERVICE_PORT")) > 0 {
```

Comment:
sure, added todo. if this change we can still use the OPENSHIFT_DEFAULT_REGISTRY to set the hostname and the port as env variable.